### PR TITLE
fix: fully manage yum confluent.repo file

### LINKED ISSUE #833: Should fully manage confluent.repo
**Describe the issue**
We are in the process of testing CP 7.0.0 in our clusters, which means that we upgrade/downgrade between CP 6.2.1 and 7.0.0. Since a dedicated client RPM-repo was introduced, it is added to confluent.repo whenever we upgrade to CP 7.0.0. But is **not** removed when downgrading to CP 6.2.1. And this is really annoying, especially after the `yum makecache` task was introduced in https://github.com/confluentinc/cp-ansible/pull/786.

I think it would be better if cp-ansible fully managed the confluent.repo file, and not patching it in multiple tasks.

**To Reproduce**
Starting with 6.2.1, upgrade to 7.0.0, then downgrade to 6.2.1. Observe that `Confluent.clients` is left in `confluent.repo`.

**Expected behaviour**
`Confluent.clients` should be removed from `confluent.repo` when downgrading, since it is not part of the desired state.

**Environment (please complete the following information):**
 - OS: RHEL
 - CP-Ansible Branch: Multiple, at least 6.2.x and 7.0.0-post

### PR DESCRIPTION
**Pull request by erikgb**
# Description

This PR makes cp-ansible fully manage the yum `confluent.repo` file using a template; to mend the version upgrade/downgrade issues described in #833. This will assure the correct/desired state on the repo-file when repositories are removed on upgrade/downgrade.

This might be considered a breaking change if any user has added additional repositories to `confluent.repo`.

Fixes #833 

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules
- [ ] Any variable changes have been validated to be backwards compatible

### CODE REVIEW

**Review by nsharma-git — APPROVED** [review]

**Comment by nsharma-git** [comment]
Hello @erikgb 
The pre check job fails with below error. 
```
13:35:57  TASK [confluent.common : Red Hat Repo Setup and Java Installation] *************
13:35:57  included: /home/jenkins/workspace/c_Contributors_cp-ansible_PR-834/roles/confluent.common/tasks/redhat.yml for zookeeper16, zookeeper26, zookeeper36, kafka-broker16, kafka-broker26, kafka-broker36, schema-registry16, kafka-connect16, ksql16, control-center16, kafka-rest16
13:35:57  
13:35:57  TASK [confluent.common : Add Confluent Repo file] ******************************
13:36:00  An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleUndefinedVariable: 'http_proxy' is undefined
13:36:00  fatal: [ksql16]: FAILED! => {"changed": false, "msg": "AnsibleUndefinedVariable: 'http_proxy' is undefined"}
13:36:00  An exception occurred during task execution. To see the full traceback, use -vvv. The error was: ansible.errors.AnsibleUndefinedVariable: 'http_proxy' is undefined
```
Would you mind taking a look on this.

File: roles/confluent.common/templates/confluent.repo.j2 [hunk 1]
```diff
@@ -0,0 +1,13 @@
+# Maintained by Ansible
+{% for name, repo in confluent_yum_repositories.items() %}
+[{{ name }}]
+async = 1
+baseurl = {{ repo.baseurl }}
+enabled = {{ repo.enabled | int }}
+gpgcheck = {{ repo.gpgcheck | int }}
+gpgkey = {{ repo.gpgkey }}
+name = {{ repo.description }}
+{% if http_proxy in proxy_env %}
```
Comment (erikgb) [inline roles/confluent.common/templates/confluent.repo.j2:10]:
```suggestion
{% if 'http_proxy' in proxy_env %}
```

**Comment by erikgb** [comment]
@nsharma-git I think I found the bug. Missing some quotes on a dict key. I will rebase and squash the commits, but please check if the error is gone.

**Comment by nsharma-git** [comment]
Thanks @erikgb The changes looks good. The PR is successful. We are good to go.
I'll wait for your new squashed commit

**Comment by erikgb** [comment]
Rebased and squashed, should be GTM. Thanks for reviewing @nsharma-git!

**Pull request merged**